lib/chef/resource.rb

```diff
@@ -1304,7 +1304,7 @@ def self.sorted_descendants
     end
     def self.inherited(child)
       super
-      @sorted_descendants = nil
+      @@sorted_descendants = nil
```

Comment:
It was wrong: that code is supposed to reset the cache on the list of descendants when you create a new subclass of `Chef::Resource`, but it's resetting the wrong variable.  The tests failed once I started using sorted_descendants for real.